server/fileCompress/app.py

```diff
@@ -0,0 +1,103 @@
+import os
+import zipfile
+import json
+import posixpath
+from pprint import pprint
+import boto3
+import shutil
+
+
+def lambda_handler(event, context):
+    # Define the constants
+    TABLE_NAME = os.environ["JOB_TABLE"]
+    BUCKET_NAME = os.environ["SRC_BUCKET"]
+    PRIMARY_KEY = "id"
+    LOOKUP_BATCH_ID = event["queryStringParameters"]["id"]
+
+    # Create table and bucket reference
+    table = boto3.resource("dynamodb").Table(TABLE_NAME)
+    bucket = boto3.resource("s3").Bucket(BUCKET_NAME)
+
+    # Get the data from DynamoDB
+    response = table.get_item(Key={PRIMARY_KEY: LOOKUP_BATCH_ID})
+
+    # Download files using reference data from DynamoDB
+    files_json = response["Item"]["files"]
+
+    # Create zip file
+    zip_file_path = os.path.join("/tmp/", "temp_zip.zip")
+    temp_zip = zipfile.ZipFile(zip_file_path, "w")
+
+    # Go through each file
+    for file in files_json:
+
+        # Get file path for s3 bucket and create file path to store in tmp
+        file_ref_s3 = file["fileRef"]
+        file_ref_tmp = "/tmp/" + file["fileRef"].split("/")[-1]
+
+        # Download the uploaded file
+        bucket.download_file(file_ref_s3, file_ref_tmp)
+
+        # Add it to the zip file in the uploaded section
+        temp_zip.write(
+            file_ref_tmp,
+            "uploaded/" + file["fileRef"].split("/")[-1],
+            compress_type=zipfile.ZIP_DEFLATED,
+        )
+
+        # Delete the upload file
+        os.remove(file_ref_tmp)
+
+        # Download the result file is it exists
+        if file["finished"]:
```

Comment:
You shouldn't need to check for this - if the file isn't finished processing then this lambda won't be called. You should instead be aborting any time a file path fails to resolve in an S3 bucket, which would handle the case of files missing as well as results missing.

Reply:
Abort the entire lambda function? Would that not happen automatically as I'm not catching the error

Reply:
But yeah, I'll get rid of the finished check. I thought there wouldn't be any harm in putting it in there

Reply:
I don't know how the abort would work, we'd probably have some error queue it would push to which would then check for integrity, but we don't have to worry about that yet (or ever possibly).